File: font-end/src/components/services/categoryService.js
import axios from "axios";

class CategoryService{

    addCategory=(categoryRequest)=>{
        return axios.post(`http://localhost:8080/category/add`,categoryRequest);
    }

    getAll=(page)=>{
        return axios.get(`http://localhost:8080/category/list/${page}`);
    }

    getTotalPage=()=>{
        return axios.get(`http://localhost:8080/category/totalPage`)
    }

    getById=(id)=>{
        return axios.get(`http://localhost:8080/category/${id}`)
    }

    update=(id,categoryRequest)=>{
        return axios.put(`http://localhost:8080/category/update/${id}`,categoryRequest);
    }

    findByAll=(name,status,page)=>{
        return axios.get(`http://localhost:8080/category/find`,{
            params:{
                name:name,
                status:status,
                page:page
            }
        })
    }

    findByAllTotalPage=(name,status)=>{
        return axios.get(`http://localhost:8080/category/findTotalPage`,{
            params:{
                name:name,
                status:status
            }
        })
    }


}

export default new CategoryService();

File: back-end/src/main/java/com/nvm/project1/service/CategoryService.java
package com.nvm.project1.service;

import com.nvm.project1.entity.Category;
import com.nvm.project1.request.CategoryRequest;
import com.nvm.project1.response.CategoryResponse;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface CategoryService {

    List<CategoryResponse> getAll();

    CategoryResponse add(CategoryRequest categoryRequest);

    List<CategoryResponse> getAllPaging(Pageable pageable);

    Double getAllTotalPage();

    CategoryResponse getById(Long id);

    CategoryResponse update(Long id,CategoryRequest categoryRequest);

    List<CategoryResponse> findByAll(String name,Boolean status,Pageable pageable);

    Double findAllTotalPage(String name,Boolean status);
}
